mdsuite/calculators/einstein_distinct_diffusion_coefficients.py

```diff
@@ -170,35 +170,110 @@ def __call__(
             species=species,
             fit_range=fit_range,
         )
-        self.time = self._handle_tau_values()
+        self.time = self._handle_tau_values() * self.experiment.units.time
 
         self.msd_array = np.zeros(self.args.data_range)  # define empty msd array
 
-    def msd_operation(self, ensemble: tf.Tensor, square: bool = True):
+    def _msd_operation(self, ds_a, ds_b) -> np.ndarray:
```

Comment:
Yes, we could either make it a free function and use it for the self and distinct diffusion, however, and as you may have seen, in order to compute the distinct values properly I actually compute the self diffusion. It would be a better idea I think to merge the two calculators into one and have the distinct calculation as an option or something. What do you think? Also, for now or for later? I would be a fan of doing this during the calculator restructure as these calculators are currently desired by some users.

Reply:
later is ok. as long as there is not too much code duplication